### PR TITLE
Chore: Cleanup package-lock.json after Gutenberg Mobile migration

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Follow-up for #22714.

This PR tries to hoist all dependencies used in the newly added React Native specific packages.

## How has this been tested?

Continues integration – GitHub actions.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-editor/package.json [hunk 1]
```diff
@@ -1,123 +1,122 @@
...
-    "@wordpress/data": "file:../data",
-    "@wordpress/edit-post": "file:../edit-post",
-    "@wordpress/element": "file:../element",
-    "@wordpress/hooks": "file:../hooks",
-    "@wordpress/i18n": "file:../i18n",
-    "@wordpress/react-native-aztec": "file:../react-native-aztec",
-    "@wordpress/react-native-bridge": "file:../react-native-bridge",
-    "fast-average-color": "^4.3.0",
-    "jed": "^1.1.1",
-    "jsc-android": "^241213.1.0",
```
Comment (gziolo) [inline packages/react-native-editor/package.json:48]: `jsc-android` isn't explicitly used in the package. It's listed as a direct dependency of `react-native` though.

File: packages/react-native-editor/package.json [hunk 2]
```diff
@@ -1,123 +1,122 @@
...
+		"@wordpress/hooks": "file:../hooks",
+		"@wordpress/i18n": "file:../i18n",
+		"@wordpress/react-native-aztec": "file:../react-native-aztec",
+		"@wordpress/react-native-bridge": "file:../react-native-bridge",
+		"fast-average-color": "^4.3.0",
+		"jed": "^1.1.1",
+		"jsdom-jscore-rn": "git+https://github.com/iamcco/jsdom-jscore-rn.git#a562f3d57c27c13e5bfc8cf82d496e69a3ba2800",
+		"metro-react-native-babel-preset": "0.57.0",
+		"metro-react-native-babel-transformer": "0.56.0",
+		"node-fetch": "^2.6.0",
```
Comment (gziolo) [inline packages/react-native-editor/package.json:51]: I updated `node-fetch` to the latest version here and in `@wordpress/e2e-test-utils`.

File: packages/react-native-editor/package.json [hunk 3]
```diff
@@ -1,123 +1,122 @@
 {
-  "name": "@wordpress/react-native-editor",
```
Comment (gziolo) [inline packages/react-native-editor/package.json:2]: I saved this file with IDE and `.editorconfig` kicked in. The best way to review is to use:
Screen Shot 2020-06-29 at 09 04 57

**Review by ntwb — APPROVED** [review]

**Comment by gziolo** [comment]
There an issue with one of the development dependencies:

Screen Shot 2020-06-29 at 09 37 01


It looks more like an issue with the `didyoumean` package configuration that is listed as licensed `Apache` in `package.json file:
https://github.com/dcporter/didyoumean.js/blob/94aecc770ada684d9209a20ade956ce2fa0ae63d/package.json#L26

In fact, it's licensed as Apache License, Version 2.0: 

https://github.com/dcporter/didyoumean.js/blob/master/LICENSE

Now the question is, whether we should extend the list of allowed values:

https://github.com/WordPress/gutenberg/blob/e421896aa6a13d2cc6e2778ea645532b693b4afb/packages/scripts/scripts/check-licenses.js#L78-L81

or explicitly exclude this package from the check? We could also change the logic that scans LICENSE files and use that as fallback when the license is incorrect in `package.json` file:

https://github.com/WordPress/gutenberg/blob/e421896aa6a13d2cc6e2778ea645532b693b4afb/packages/scripts/scripts/check-licenses.js#L113

It probably would allow us also to remove all the variations people use to describe Apache v2.0 license.

**Comment by pento** [comment]
Changing the logic of `check-licenses.js` to fall back to the LICENSE file if it can't match the value in `package.json` seems reasonable to me. The LICENSE file is probably the canonical source, the value in `package.json` is just a shortcut.

**Comment by gziolo** [comment]
Updated the logic in `check-licenses` with 96365c2. I was able to remove some of the listed licenses types that seem like a mix of multiple licenses or wrong classification.

**Review by ceyhun — APPROVED** [review]
Tested Android and iOS demo apps and they work fine. Thank you for this!

**Pull request merged**